torch/nn/parallel/distributed.py

```diff
@@ -1797,6 +1797,12 @@ def _set_params_and_buffers_to_ignore_for_model(
         # during synchronization. It will be removed when the API is finalized
         # as part of addressing https://github.com/pytorch/pytorch/issues/43690.
         module._ddp_params_and_buffers_to_ignore = params_and_buffers_to_ignore
+        for name, param in module.named_parameters():
+            if name in params_and_buffers_to_ignore:
+                param._ddp_ignored = True
+        for name, buffer in module.named_buffers():
```

Comment:
@mrshenli i thought buffers by definition do not require grad, and therefore ddp ignores them by default?

if not i should update the logic in DDPOptimizer accordingly.

Reply:
Buffers are by default broadcast right before the forward pass if  broadcast_buffers=True
is passed to DDP constructor. This is true by default.
But then, if the buffer appear on the igore_parameters field, it's not part of the broadcast.
I think buffers shouldn't count for the purposes of splitting the model since we're not syncing them after the backward pass.